### PR TITLE
feat(auth): add auth endpoints

### PR DESCRIPTION
**Pull request by wass3r**

### CODE REVIEW

File: server/authentication.go [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+	"github.com/gin-gonic/gin"
+	"github.com/go-vela/types"
+	"github.com/go-vela/types/constants"
+	"github.com/go-vela/types/library"
+)
+
+const (
+	// TokenRefreshResp represents a JSON return for a token refresh
+	TokenRefreshResp = `{
```
[the remark this answers is not in the document]
Comment (wass3r) [reply server/authentication.go:20]: linter comment added

File: server/authentication.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/go-vela/types"
+	"github.com/go-vela/types/constants"
+	"github.com/go-vela/types/library"
+)
+
+const (
+	// TokenRefreshResp represents a JSON return for a token refresh
```
[the remark this answers is not in the document]
Comment (wass3r) [reply server/authentication.go]: fixed

**Review by matt-fevold — APPROVED** [review]

**Review by jbrockopp — APPROVED** [review]
LGTM

**Review by kneal — APPROVED** [review]
LGTM 🐬

**Pull request merged**